microsoft-edge/progressive-web-apps-chromium/how-to/widgets.md

```diff
@@ -0,0 +1,389 @@
+
+| Method | Description | Parameters | Return value |
+|:--- |:--- |:--- |:--- |
+| `getByTag(tag)` | Get a Widget by tag | The Widget tag | A Promise that resolves to the [Widget object](#widget-object) that matches the tag, or `undefined`. |
+| `getByInstanceId(id)` | Get a Widget by instance ID | The Widget instance ID | A Promise that resolves to the corresponding [Widget object](#widget-object), or undefined.
+| `getByHostId(id)` | Get Widgets by host ID  | The host ID | An array of [Widget objects](#widget-object) found in that host.
+| `matchAll(options)` | Get Widgets by matching options | A [WidgetOptions object](#widgetoptions-object) | A Promise that resolves to an array of [Widget objects](#widget-object) that match the `options` criteria. |
+| `updateByInstanceId(id, payload)` | Update a Widget by instance ID  | The instance ID, and a [WidgetPayload object](#widgetpayload-object) | A Promise that resolves to `undefined` or `Error`.
+| `updateByTag(tag, payload)` | Update a Widget by tag | The Widget tag, and a [WidgetPayload object](#widgetpayload-object) | A Promise that resolves to undefined or Error.
+| `removeByInstanceId(id)` | Remove a Widget by instance ID | The instance ID | A Promise that resolves to undefined or Error.
+| `removeByTag()` | Remove a Widget by tag | The Widget tag | A Promise that resolves to undefined or Error.
+
+### Widget object
+
+Each Widget is represented as a `Widget` object, which contains the following properties:
+
+* `installable`: A Boolean indicating wither the Widget is installable.
+* `definition`: A [WidgetDefinition object](#widgetdefinition-object).
+* `instances`: An array of [WidgetInstance objects](#widgetinstance-object) representing the current state of each instance of the Widget.
+
+### WidgetOptions object
+
+When using `matchAll(options)` to get multiple Widgets, a `WidgetOptions` object is necessary to filter which Widgets to return. The `WidgetOptions` object contains the following properties, all of which are optional:
+
+* `installable`: A Boolean that indicates if the Widget is considered installable.
+* `installed`: A Boolean that indicates if the Widget has been installed in the Widget host.
+* `tag`: The tag of the Widget.
+* `instance`: The Widget instance.
+* `host`: The Widget host ID.
+
+### WidgetPayload object
+
+When creating or updating a Widget instance, the service worker must send the data that's necessary to populate the Widget. This data is called the _payload_.  The payload includes template-related data and content-related data.  The `WidgetPayload` object contains the following properties:
+
+* `template`: The name of the template to use.
+* `data`: The data to use with the Widget template, as a `String`.  This data can be stringified JSON data.
```

Comment:
Currently, the `WidgetPayload` **only** has `data`. The other two are not yet supported.

Settings will be added post-SV2 when Widget Setting support is available, and it's unclear if/when `template` will be added.